tests-kernels/__init__.py

```diff
@@ -0,0 +1,38 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/master/LICENSE
+
+import ctypes
+import os
+import platform
+
+import pkg_resources
+
+if platform.system() == "Windows":
+    name = "awkward-cpu-kernels.dll"
+elif platform.system() == "Darwin":
+    name = "libawkward-cpu-kernels.dylib"
+else:
+    name = "libawkward-cpu-kernels.so"
+
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+TOP_DIR = os.path.join(CURRENT_DIR, "..")
+
+CPU_KERNEL_SO = None
+for root, _, files in os.walk(TOP_DIR):
+    for filename in files:
+        if filename == name:
+            CPU_KERNEL_SO = os.path.join(root, filename)
+            break
+
+if CPU_KERNEL_SO is None:
+    CPU_KERNEL_SO = pkg_resources.resource_filename("awkward1", name)
```

Comment:
If the second method (`pkg_resources.resource_filename`) always works, it's better than searching from the `TOP_DIR`. The active `awkward1` Python package would usually be located in a different directory than the tests; this is true in CI because they're built with `pip install .` rather than `python localbuild.py`. So if you can remove everything except for the last line and it still works, that's the better way to go.

Reply:
But then I think, someone who built it using `python localbuild.py` instead of with `pip` would not be able to run tests?

Reply:
I'd expect `pkg_resources.resource_filename("awkward1", name)` to point to the version in the nearby `awkward1` directory in that case. If it doesn't work and the explicit search is needed, it's still probably better to check `pkg_resources` first, since you don't want a nearby directory to overshadow the directory corresponding to `import awkward1`.